json-rpc/types/src/views.rs

```diff
@@ -115,7 +117,7 @@ impl AccountView {
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub struct EventView {
-    pub key: BytesView,
+    pub key: EventKey,
```

Comment:
I don't think this represents what the final result we defined for the JSON-RPC response type.
I'd rather to make BytesView to string, because that's what is the right type for the data.

Reply:
But that's exactly what it is? An `EventKey` hex encoded.

Reply:
View types are the definition of the json type in current state, it should represent json type instead of rust type.
You are right with your change the final json result is hex encoded string, but it can be different result depending on how you render the type into json, which is hard to understand by reading this definition, which should just tell you what is json type is.

Reply:
It will be fine if you only use this for a client, which deserializes json data into Rust types.
But it is not good for defining what is server JSON response type, because it makes harder for reader to understand what is json type looks like, and current view types have here and there don't match json type already, which is what I'd like to improve instead of making it worse.

Reply:
I disagree that the type the server uses should match the rendered Json. Doing things this way, stricter definition of types, is a way to use the type system to help stop you from making mistakes. Instead of any "arbitrary string" this particular field should be an `EventKey` and the type system will make sure that's the case. That's the biggest benefit we have from using a strictly typed language.

Reply:
that's where confusing comes from, as json rendered result should be string instead of `EventKey`

Reply:
JSON is pretty low level and only has a few basic types. I don't think we should limit the high level API to only the types JSON supports. This makes using the API very inconvenient because you must constantly deal with serialization yourself.

You may be right that the programmer will want to understand how it serializes under the hood, but the serde annotations and/or doc comments should be able to make that clear if it is confusing.

Does the Python SDK use higher level types and manage serialization or does it just treat everything as strings for the programmer to deal with like here?

Reply:
synced offline, we will make sure have test to against a json-rpc type definition to ensure server response matches the definition.